### PR TITLE
Port the support for ':profile stop' and ':profile dump' commands from Neovim

### PR DESCRIPTION
**Pull request by yegappan**
This PR includes the following changes made in Neovim:

Author: Marco Hinz <mh.codebro@gmail.com>
Commit: https://github.com/neovim/neovim/commit/1902ee52ef60d96f2ad043687a1e86bcf7be8aed
Commit: https://github.com/neovim/neovim/commit/588bf76650d6d645403e70437c649e7759430c05

### CODE REVIEW

File: runtime/doc/repeat.txt [hunk 1]
```diff
@@ -1071,6 +1074,9 @@ For example, to profile the one_script.vim script file: >
 		after this command.  A :profile command in the script itself
 		won't work.
 
+:prof[ile] dump
+               Don't wait until exiting Vim and write the current state of
+               profiling to the log immediately.
 
 :profd[el] ...						*:profd* *:profdel*
```
Comment (dpelle) [inline runtime/doc/repeat.txt:1086]: Slightly off-topic, but I don't find the doc of `:profdel` clear.

File: runtime/doc/repeat.txt [hunk 2]
```diff
@@ -1048,6 +1048,9 @@ For example, to profile the one_script.vim script file: >
 		If {fname} already exists it will be silently overwritten.
 		The variable |v:profiling| is set to one.
 
+:prof[ile] stop
+               Write the logfile and stop profiling.
```
Comment (dpelle) [inline runtime/doc/repeat.txt]: I have not tried yet, but what happens if I call `:profile stop` twice in a row?

I also see that behavior of calling `:profile start …` twice is not documented (already the case prior to your change).

Also what happens if we call `:profile dump` without prior call to `:profile start …`?

**Comment by vim-ml** [comment]
Hi,

On Thu, Apr 7, 2022 at 12:46 AM Dominique Pellé ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In runtime/doc/repeat.txt
> <https://github.com/vim/vim/pull/10107#discussion_r844822293>:
>
> > @@ -1071,6 +1074,9 @@ For example, to profile the one_script.vim script file: >
>
>  		after this command.  A :profile command in the script itself
>
>  		won't work.
>
>
>
> +:prof[ile] dump
>
> +               Don't wait until exiting Vim and write the current state of
>
> +               profiling to the log immediately.
>
>
>
>  :profd[el] ...						*:profd* *:profdel*
>
>
> Slightly off-topic, but I don't find the doc of :profdel clear.
> ------------------------------
>
> In runtime/doc/repeat.txt
> <https://github.com/vim/vim/pull/10107#discussion_r844822380>:
>
> > @@ -1048,6 +1048,9 @@ For example, to profile the one_script.vim script file: >
>
>  		If {fname} already exists it will be silently overwritten.
>
>  		The variable |v:profiling| is set to one.
>
>
>
> +:prof[ile] stop
>
> +               Write the logfile and stop profiling.
>
>
> I have not tried yet, but what happens if I call :profile stop twice in a
> row?
>

The second time, the error message "E750: First use ":profile start
{fname}" will be displayed.


> I also see that behavior of calling :profile start … twice is not
> documented (already the case prior to your change).
>
> Also what happens if we call :profile dump without prior call to :profile
> start …?
>
>
>
If ":profile dump" is called without a start, then the error message "E750:
First use ":profile start {fname}"
will be displayed.

- Yegappan

**Comment by brammool** [comment]
Just to be clear:
- After "profile dump" profiling continues as before, adding to the already collected stats?
- After "profile stop" one can use "profile start" again and it will use a clean slate?  Thus doing "profile start / some work / profile stop" twice would give very similar results?
Would be good to make this clear in the help.

**Comment by vim-ml** [comment]
Hi Bram,

On Thu, Apr 7, 2022 at 10:58 AM Bram Moolenaar ***@***.***>
wrote:

> Just to be clear:
>
>    - After "profile dump" profiling continues as before, adding to the
>    already collected stats?
>
>
Yes. After executing the ":profile dump" command, the profiling continues
and adds to the
already collected stats.


>
>    - After "profile stop" one can use "profile start" again and it will
>    use a clean slate? Thus doing "profile start / some work / profile stop"
>    twice would give very similar results?
>
>
Yes. The new tests already use the above sequence of commands.


>
>    - Would be good to make this clear in the help.
>
>
>
Will update the help to reflect this.

Regards,
Yegappan

**Pull request closed without merging**